xcode/Subconscious/Shared/Services/NoosphereService.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  NoosphereService.swift
+//  Subconscious (iOS)
+//
+//  Created by Gordon Brander on 1/10/23.
+//
+//  Noosphere lifecycle:
+//  1. Initialize namespace (once ever)
+//  2. Create key (once per key)
+//  3. Create sphere for key (once per key)
+//  4. Retreive sphere identity from sphere creation receipt, then store (once, upon sphere creation)
+//  5. Retrieve sphere mnemonic from sphere creation receipt. Display to user, then discard (once, upon sphere creation).
+//  6. Open sphere file system (on-demand)
+
+import Foundation
+import SwiftNoosphere
+
+public enum NoosphereServiceError: Error {
+    case foreignError(String)
+}
+
+public struct SphereConfig {
+    var identity: String
+    var mnemonic: String
+}
+
+/// A Swift Wrapper for Noosphere.
+/// This class is designed to follow RAII pattern https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization.
+/// Memory is automatically freed when the class is deallocated.
+public final class NoosphereService {
+    /// Create and configure a user and sphere
+    /// - Important: the identity should be persisted
+    /// - Important: the mnemonic should be displayed to user and then discarded.
+    /// - Returns: a SphereConfig containing the sphere identity and mnemonic
+    ///
+    /// What it does:
+    /// - Initializes a namespace
+    /// - Creates a key
+    /// - Creates a sphere for that key
+    public static func create(
+        ownerKeyName: String,
+        globalStoragePath: String,
+        sphereStoragePath: String
+    ) throws -> SphereConfig {
+        guard let noosphere = ns_initialize(
+            globalStoragePath,
+            sphereStoragePath,
+            nil
+        ) else {
+            throw NoosphereServiceError.foreignError(
+                "Failed to get pointer for namespace"
+            )
+        }
+        defer {
```

Comment:
Defer runs immediately after this scope exits, even if the function throws https://www.hackingwithswift.com/new-syntax-swift-2-defer